docs/feature-debugging.md

```diff
@@ -0,0 +1,83 @@
+
+    <center><img src="../img/feature-debugging-3.png" style="height: 100px;"/></center>
+
+Selecting one or the other will instruct the Debug pane to show details for that "side" of the application e.g. breakpoints, call stack etc.
+
+### Setting breakpoints
+To set a breakpoint, simply highlight the line of code that you wish to stop on and hit `F9`, or click on the left-side gutter of the code window; a red dot will appear to show the breakpoint is set. You can see more on VS Code breakpoints [here](https://code.visualstudio.com/docs/editor/debugging#_breakpoints).
+
+Here you can see a breakpoint that has been hit in the `update` method of the sample application that is bundled with the SAFE template.
+
+<center><img src="../img/feature-debugging-4.png" style="height: 200px;"/></center>
+
+### Viewing debug output
+Debug output from Chrome will automatically be sent to the integrated Debug Console. This is especially useful for Elmish applications, as you can interrogate message and model updates directly in the IDE.
+
+<center><img src="../img/feature-debugging-6.png" style="height: 200px;"/></center>
+
+> Note: If you observe error messages regarding `SocketProtocolError` (see below), you can eliminate them by installing the Redux Dev Tools in the launched Chrome instance, as mentioned earlier. However, these errors can also safely be ignored. 
+
+<center><img src="../img/feature-debugging-5.png" style="height: 175px;"/></center>
+
+## Restarting your application
+In the standard launch mode seen above, hot module reloading is enabled on the client, so changes that are made to your Fable application will occur immediately.
+
+However, in order to support debugging, watch mode is not enabled on the server; server changes require a restart of the server debugging session: Having ensured that Debug Server is selected in the floating Debug control, select the Restart icon or hit `CTRL + SHIFT + F5`. The server will stop, rebuild and restart with the debugger automatically re-attached.
+
+## Advanced options
+The SAFE template comes with several other modes of operation for debugging:
+
+### Full watch mode
+You can elect to use "watch" mode on both client *and* server by choosing the **Watch SAFE app** option from the Debug Launcher. Here, changes to any server file will automatically initiate a restart of it. However, whilst you will still be able to debug the client, in this mode there is no capability for server-side debugging.
+
+### Debug Server or Client only
+You can elect to launch and debug only the client or server in isolation. This may be useful if you are running the other component separately e.g. through FAKE, the `dotnet` command line tool directly or via Visual Studio. You can initiate this by choosing either **Debug Client** or **Debug Server** as required.
+
+### Build Server
+You can initiate a restore and rebuild of the server by hitting `CTRL + SHIFT + B` and selecting **Build Server** from the drop-down.
```

Comment:
Maybe we should add a "Known issues" / "Troubleshooting" section at the very end listing:
1. issue with node process staying alive after debug session (with link to GH issue)
1. SocketProtocolError with ReduxDevTools
1. Chrome window starting before it's actually useful